### PR TITLE
doc: fix javadoc in TypeFactory.java

### PR DESCRIPTION
**Pull request by zielint0**
SonarQube: *Remove this unnecessary cast to "CtElement".*

### CODE REVIEW

**Comment by monperrus** [comment]
We have two different PRs here (typo and cast). They should be separated in two.

File: src/main/java/spoon/reflect/factory/TypeFactory.java [hunk 1]
```diff
@@ -282,7 +282,7 @@ public TypeFactory() {
 	}
 
 	/**
-	 * Creates a reference to a one-dimension array of given type.
```
Comment (monperrus) [inline src/main/java/spoon/reflect/factory/TypeFactory.java]: "a one" is correct, see https://www.quora.com/Which-is-correct-a-one-or-an-one

**Comment by zielint0** [comment]
> We have two different PRs here (typo and cast). They should be separated in two.

Cast is in https://github.com/INRIA/spoon/pull/2688

File: src/main/java/spoon/reflect/factory/TypeFactory.java (hunk 1 above)
Comment (zielint0) [reply src/main/java/spoon/reflect/factory/TypeFactory.java]: @monperrus 
That's right. Thank you for your careful review!

**Pull request merged**